src/transactions/TransactionFrame.cpp

```diff
@@ -1329,6 +1334,143 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
     return false;
 }
 
+bool
+TransactionFrame::applyLifetimeBumps(Application& app, AbstractLedgerTxn& ltx)
+{
+    if (!isSoroban())
+    {
+        return true;
+    }
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    LedgerTxn ltxTx(ltx);
+    auto const& resources = sorobanResources();
+    auto lcl = app.getLedgerManager().getLastClosedLedgerNum();
+    auto const& networkConfig =
+        app.getLedgerManager().getSorobanNetworkConfig(ltx);
+
+    auto maxExpirationLedger = lcl + networkConfig.maximumEntryLifetime();
+    auto minRestorableExpirationLedger =
+        lcl + networkConfig.minimumRestorableEntryLifetime();
+    auto minTempExpirationLedger =
+        lcl + networkConfig.minimumTempEntryLifetime();
+    auto autoBumpLedgers = networkConfig.autoBumpLedgers();
+
+    auto autoBumpEnabled = [](LedgerEntry const& le) {
+        return le.data.type() != CONTRACT_DATA ||
+               (le.data.contractData().body.data().flags &
+                ContractDataFlags::NO_AUTOBUMP);
+    };
+
+    auto bump = [&](LedgerEntry& le, auto const& bumpLedgers) {
+        // Use uint64_t do avoid potential overflows from footprint values then
+        // cast down later
+        uint64_t newExpiration = getExpiration(le);
+        if (bumpLedgers)
+        {
+            newExpiration += *bumpLedgers;
+        }
+        else
+        {
+            if (autoBumpEnabled(le))
+            {
+                newExpiration += autoBumpLedgers;
+            }
+        }
+
+        auto minExpirationLedger = isTemporaryEntry(le.data)
+                                       ? minTempExpirationLedger
+                                       : minRestorableExpirationLedger;
+
+        if (newExpiration > maxExpirationLedger)
+        {
+            newExpiration = maxExpirationLedger;
+        }
+        else if (newExpiration < minExpirationLedger)
+        {
+            newExpiration = minExpirationLedger;
```

Comment:
We have to enforce the minimum expiration to prevent rent gaming since we can't actually start expiring UNIQUE and RESTORABLE entries until level 6 of the BucketList. This can cause exceeding the fee, but the thinking is that since rent payments are coming out of refundable fee, users should be able to be overly cautious and put in a little extra XLM for cases such as this. However we only need to enforce this on writes, not for the read-only entries. I think currently this applies the minimum to both read only and writes, so I'll change that, but keep the minimum as it is for modified entries.